### PR TITLE
[ci] Allow using local wheels to run release tests. 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

This allows one to directly run release tests on a local wheel built so one doesn't need to wait for PR wheel build. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by rkooo567** [comment]
Will leave it up to @krfricke

File: release/ray_release/wheels.py [hunk 1]
```diff
@@ -130,6 +133,36 @@ def parse_wheels_filename(
...
 
+def upload_to_s3(src_path: str, bucket: str, key_path: str) -> Optional[str]:
+    s3_client = boto3.client("s3")
+    try:
+        s3_client.upload_file(Filename=src_path, Bucket=bucket, Key=key_path)
+    except ClientError as e:
+        logger.warning(f"Failed to upload to s3:  {e}")
+        return None
+
+    return f"https://{bucket}.s3.us-west-2.amazonaws.com/{key_path}"
```
Comment (krfricke) [inline release/ray_release/wheels.py:144]: Can we move this to `aws.py`?

File: release/ray_release/wheels.py [hunk 2]
```diff
@@ -130,6 +133,36 @@ def parse_wheels_filename(
...
+
+
+def get_ray_wheels_url_from_local_wheel(ray_wheels: str) -> Optional[str]:
+    """Upload a local wheel file to S3 and return the downloadable URI"""
+    import os
+    from datetime import datetime
```
Comment (krfricke) [inline release/ray_release/wheels.py:150]: Let's move the imports to the top

**Review by krfricke — COMMENTED** [review]
Looks good, minor nits.

Have you tested this locally (does it work)?

**Comment by rickyyx** [comment]
> Looks good, minor nits.
> 
> Have you tested this locally (does it work)?

Yep, I have been using it:

image

You could also see the bucket here: https://s3.console.aws.amazon.com/s3/buckets/ray-release-manual-wheels?region=us-west-2&tab=objects

**Review by krfricke — APPROVED** [review]
Looks good to me, thanks! cc @can-anyscale for reference: This is for local CLI mode (not buildkite) so I think we can go ahead for now. It also only affects the buildfarm, not the actual cluster where it's running on, so it should be compatible with gce.

**Comment by can-anyscale** [comment]
verynice

**Pull request merged**